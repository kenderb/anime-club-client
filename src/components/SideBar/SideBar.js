import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PropType from 'prop-types';
import { logoutUser } from '../../actions';
import './SideBar.style.css';

const SideBar = ({ logoutUser, user }) => {
  const handleonclick = () => {
    logoutUser();
  };
  return (
    <aside>
      <ul className="side-menu">
        <li>
          <img src="" alt={user.name} />
          {user.name}
        </li>
        <li>
          <Link to="/favorites">My favorites</Link>
        </li>
        <li>
          <button type="button" onClick={handleonclick}> Logout </button>
        </li>
      </ul>
    </aside>
  );
};

const mapStateToprops = state => ({
  user: state.user,
});
SideBar.propTypes = {
  logoutUser: PropType.func.isRequired,
  user: PropType.instanceOf(Object).isRequired,
};

export default connect(mapStateToprops, { logoutUser })(SideBar);
