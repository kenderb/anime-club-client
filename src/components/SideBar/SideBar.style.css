.side-menu {
  z-index: 1;
  list-style: none;
  padding: 1em;
  width: 14em;
  position: fixed;
  top: 0;
  height: 100vh;
  flex-direction: column;
  background-color: var(--bg-color);
  box-shadow: rgba(129, 129, 129, 0.356) 10px 5px 10px;
}

.side-menu > li {
  margin-bottom: 1em;
}

.side-menu a {
  text-decoration: none;
  color: black;
}

.side-menu button {
  font-size: 1em;
  border: none;
  padding: 1em 0;
}

.user-container {
  justify-content: center;
  flex-direction: column;
}

.separator {
  margin-top: auto;
  height: 1px;
  width: 100%;
  background-color: #e9e9e9;
}

.bg-black {
  background-color: #00000050;
  position: fixed;
  top: 0;
  height: 100vh;
  width: 100vw;
}

.d-none {
  display: none;
}

.user-image {
  opacity: 0.3;
  width: 8em;
  padding: 1em;
  margin: 1em;
}

.user-container {
  align-items: center;
  margin-bottom: 1em;
}

.active-mark {
  padding-left: 0.3em;
  background-color: #eeeeee9a;
  border-left: var(--primary-color) solid 4px;
}
