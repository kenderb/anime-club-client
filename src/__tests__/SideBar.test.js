import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { create } from 'react-test-renderer';
import { createStore, applyMiddleware } from 'redux';
import { BrowserRouter } from 'react-router-dom';
import combineReducers from '../reducers';
import SideBar from '../components/SideBar';

test('SideBar component renders correctly', () => {
  const store = createStore(combineReducers, applyMiddleware(thunk));
  const tree = create(
    <Provider store={store}>
      <BrowserRouter>
        <SideBar />
      </BrowserRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
