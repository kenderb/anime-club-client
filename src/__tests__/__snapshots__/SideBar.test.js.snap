// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar component renders correctly 1`] = `""`;
